components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -77,15 +120,19 @@ case object NotBlankParameterValidator extends ParameterValidator {
 
 case class FixedValuesValidator(possibleValues: List[FixedExpressionValue]) extends ParameterValidator {
 
-  override def isValid(paramName: String, value: String, label: Option[String])(
+  override def isValid(paramName: String, value: Any, label: Option[String])(
       implicit nodeId: NodeId
   ): Validated[PartSubGraphCompilationError, Unit] = {
+    // FIXME: we should properly evaluate `possibleValues`
     val values = possibleValues.map(possibleValue => possibleValue.expression)
 
-    if (StringUtils.isBlank(value) || values.contains(value))
-      valid(())
-    else
-      invalid(InvalidPropertyFixedValue(paramName, label, value, possibleValues.map(_.expression)))
+    value match {
+      case null                                  => valid(())
+      case s: String if values.contains(s"'$s'") => valid(())
```

Comment:
Fixes values are not only for strings. It was used with many other types - take a look at `AggregateHelper.SIMPLE_EDITOR`. Maybe we need to do a similar thing as we did in `SpelExpressionParameterValidator`?

Reply:
But this is just an editor, not a validator. Dual editor - to be precise, which allows user to write more complex aggregators, like `#AGG.map({last: #AGG.last, list: #AGG.list})`. The real (maybe even philosophical) question is - should we use **FixedValues**Validator for such complex objects? It was easy to compare them earlier - just check if expression strings are the same. Actually, aggregators can't even be compared:
![image](https://github.com/TouK/nussknacker/assets/3193882/d2b7b7c5-f689-42ad-91fa-f36fca854bb9)

Maybe validator API needs some further enhancement? Maybe we should also pass the raw expression?

Reply:
As I added `expression` to `isValid` method, this validator now works the same as before.

Reply:
It doesn't work the same as we discussed offline